ra2ce/analysis/analysis_collection.py

```diff
@@ -33,7 +36,28 @@
 class AnalysisCollection:
     damages_analyses: list[AnalysisDamagesProtocol] = field(default_factory=list)
     losses_analyses: list[AnalysisLossesProtocol] = field(default_factory=list)
-    adaptation_analysis: AnalysisDamagesProtocol = None
+    adaptation_analysis: Adaptation = None
+
+    def of_type(self, analysis_type: Type[AnalysisProtocol]) -> list[AnalysisProtocol]:
+        """
+        Method emulating the `OfType<T>` built-in .Net equivalent to retrieve
+        the analyses matching the provided type.
+
+        Args:
+            analysis_type (Type[AnalysisProtocol]): The requested analysis type.
+
+        Returns:
+            list[AnalysisProtocol]: All the analyses matching the provided type.
+        """
+        if issubclass(analysis_type, Adaptation):
+            # Adaptation implements `AnalysisDamagesProtocol`,
+            # so it needs to be checked first.
+            return [self.adaptation_analysis]
+        if issubclass(analysis_type, AnalysisLossesProtocol):
+            return self.losses_analyses
+        if issubclass(analysis_type, AnalysisDamagesProtocol):
+            return self.damages_analyses
```

Comment:
Related to #611: as Damages and Adaptation implement this protocol, we could rename it to AnalysisProtocol.

Reply:
I do not think so, we are checking whether the concrete class implements this protocol, if you check against a higher level protocol then all of the three cases will return `True`. 
If you want all of them to have the same interface (because they have no differences), then we should check here whether they implement the most "independent" yet generic base class possible e.g.:
- `Adaptation` -> `Adaptation`, 
- `AnalysisLossesProtocol` -> `LossesBase`,
- `AnalysisDamagesProtocol` -> `Damages`.

Reply:
Agree. Let's discuss later in the context of #611 (if we get to it)